### PR TITLE
Adding back an exported Analyze(..) API for scorch & upsidedown

### PR DESCRIPTION
**Pull request by abhinavdangeti**
Note that these APIs have different signatures for the two.

### CODE REVIEW

**Review by mschoch — CHANGES_REQUESTED** [review]
Woah this is wrong.  At best, we add Analyze() back, **just** as a wrapper around the analyze() method.  We should be reintroducing any of these types.  There should be no index/analysis package?  what is going on here?

**Comment by abhinavdangeti** [comment]
Ah shoot, you're right. There's an AnalyzedTokenFrequencies() API in bleve_index_api that I can use. I'm going to overwrite this patch, one sec.

**Review by mschoch — COMMENTED** [review]
better, but I'd still like a clear explanation of how this is used before we add it back

**Comment by mschoch** [comment]
Also, that test is failing and the name of test doesn't have the "_id" change we put in.  Is this change not based on the 2.x branch?

**Comment by abhinavdangeti** [comment]
> Also, that test is failing and the name of test doesn't have the "_id" change we put in. Is this change not based on the 2.x branch?

It is. Seems like this is breaking on some builds only.

**Comment by mschoch** [comment]
But, I changed the name of the test to include "_id" and the name of the test printed doesn't have it.

**Comment by mschoch** [comment]
Oh I see, it's actually a different test, with a similar name.  We basically have a bunch of these and will just find them as they fail...

File: test/tests/sort/searches.json [hunk 1]
```diff
@@ -170,14 +170,14 @@
 		}
 	},
   {
-    "comment": "sort by age, descending, missing first",
+    "comment": "sort by age, descending, missing first, _id, ascending",
```
Comment (sreekanth-cb) [inline test/tests/sort/searches.json:173]: I guess this is for making the test work for both scorch and upsidedown?

**Review by mschoch — APPROVED** [review]

**Pull request merged**